packages/compass-connections/src/connections-manager.ts

```diff
@@ -157,26 +224,38 @@ export class ConnectionsManager extends EventEmitter {
         dataService.addReauthenticationHandler(this.reAuthenticationHandler);
       }
 
-      this.connectionAttempts.delete(connectionInfo.id);
-      this.dataServices.set(connectionInfo.id, dataService);
+      this.connectionAttempts.delete(adjustedConnectionInfoForConnection.id);
+      this.dataServices.set(
+        adjustedConnectionInfoForConnection.id,
+        dataService
+      );
+
       this.updateAndNotifyConnectionStatus(
-        connectionInfo.id,
+        adjustedConnectionInfoForConnection.id,
         ConnectionsManagerEvents.ConnectionAttemptSuccessful,
-        [connectionInfo.id, dataService]
+        [adjustedConnectionInfoForConnection.id, dataService]
       );
+
+      dataService.on?.('connectionInfoSecretsChanged', () => {
+        onDatabaseSecretsChange(
+          adjustedConnectionInfoForConnection,
+          dataService
+        );
+      });
```

Comment:
Notification should probably happen after we did all the work, can we move this up a bit?